File: pubnub-java/src/test/java/com/pubnub/api/endpoints/files/TestsWithFiles.java
package com.pubnub.api.endpoints.files;

import org.junit.rules.TemporaryFolder;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.util.ArrayList;
import java.util.Collections;

@SuppressWarnings("UnstableApiUsage")
public interface TestsWithFiles {
    TemporaryFolder folder = new TemporaryFolder();

    TemporaryFolder getTemporaryFolder();

    default File getTemporaryFile(String filename, String... content) {
        try {
            File file = getTemporaryFolder().newFile(filename);
            ArrayList<String> lines = new ArrayList<>();
            Collections.addAll(lines, content);
            Files.write(file.toPath(), lines);
            return file;
        } catch (IOException ex) {
            //fail
            throw new RuntimeException(ex.getMessage(), ex);
        }
    }

}

File: src/test/kotlin/com/pubnub/api/legacy/endpoints/files/TestsWithFiles.kt
package com.pubnub.api.legacy.endpoints.files

import org.junit.rules.TemporaryFolder
import java.io.File
import java.io.IOException
import java.nio.file.Files
import java.util.ArrayList
import java.util.Collections

interface TestsWithFiles {
    val temporaryFolder: TemporaryFolder

    fun getTemporaryFile(filename: String, vararg content: String?): File {
        return try {
            val file = temporaryFolder.newFile(filename)
            val lines = ArrayList<String>()
            Collections.addAll<String>(lines, *content)
            Files.write(file.toPath(), lines)
            file
        } catch (ex: IOException) {
            // fail
            throw RuntimeException(ex.message, ex)
        }
    }

    companion object {
        val folder = TemporaryFolder()
    }
}
